pkg/infra/log/text/text_logger.go

```diff
@@ -0,0 +1,40 @@
+package text
+
+import (
+	"encoding"
+	"fmt"
+	"io"
+	"reflect"
+
+	gokitlog "github.com/go-kit/log"
+)
+
+type textLogger struct {
+	w io.Writer
+}
+
+// NewTextLogger similar to gokitlog.NewLogfmtLogger
+// but converts unsupported types to string
+func NewTextLogger(w io.Writer) gokitlog.Logger {
+	return &textLogger{w}
+}
+
+func (l textLogger) Log(keyvals ...interface{}) error {
+	for i, val := range keyvals {
+		switch val.(type) {
+		case nil, string, []byte, encoding.TextMarshaler, error, fmt.Stringer:
+		default:
+			rvalue := reflect.ValueOf(val)
+			switch rvalue.Kind() {
+			case reflect.Array, reflect.Chan, reflect.Func, reflect.Map, reflect.Slice, reflect.Struct:
+				keyvals[i] = fmt.Sprintf("%+v", val)
+			default:
+			}
```

Comment:
Do we really need these two paths? It seems like we could get rid of a semi-slow reflection step by just doing:

```suggestion
			keyvals[i] = fmt.Sprintf("%+v", val)
```

If we do, I think it can be streamlined a bit by using TypeOf instead of ValueOf?

```suggestion
			switch reflect.TypeOf(val) {
			case reflect.Array, reflect.Chan, reflect.Func, reflect.Map, reflect.Slice, reflect.Struct:
				keyvals[i] = fmt.Sprintf("%+v", val)
			default:
			}
```

Reply:
I think that we cannot get completely rid of the reflection because [there are some other cases that do not result to error](https://github.com/go-logfmt/logfmt/blob/99455b83edb21b32a1f1c0a32f5001b77487b721/encode.go#L158) and it would be better to leave them intact.

but I'll apply the `reflect.TypeOf()` suggestion